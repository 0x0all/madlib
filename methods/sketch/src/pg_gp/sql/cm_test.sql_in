set search_path to "$user",public,MADLIB_SCHEMA;
-- Basic methods
select cmsketch_count(cmsketch(i),5) from generate_series(1,10000) as T(i);
select cmsketch_rangecount(cmsketch(i),1,1025) from generate_series(1,10000) as T(i);
select cmsketch_rangecount(cmsketch(i),1,200) from generate_series(1,10000) as R(i);
select cmsketch_width_histogram(cmsketch(i), min(i), max(i), 4) from generate_series(1,10000) as R(i);
select min(i),
       cmsketch_centile(cmsketch(i), 25, count(i)) AS quartile1, 
       cmsketch_centile(cmsketch(i), 50, count(i)) AS quartile2, 
       cmsketch_median(cmsketch(i), count(i)) AS median,
       cmsketch_centile(cmsketch(i), 75, count(i)) AS quartile3,
       max(i) 
  from generate_series(1,10000) as R(i);
select cmsketch_depth_histogram(cmsketch(i), 4) from generate_series(1,10000) as R(i);
-- test for all-NULL column
select cmsketch_count(cmsketch(NULL), 5) from generate_series(1,10000) as R(i) where i < 0;

