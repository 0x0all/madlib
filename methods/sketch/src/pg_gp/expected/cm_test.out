set search_path to "$user",public,madlib;
-- Basic methods
select cmsketch_count(cmsketch(i),5) from generate_series(1,10000) as T(i);
 cmsketch_count 
----------------
              5
(1 row)

select cmsketch_rangecount(cmsketch(i),1,1025) from generate_series(1,10000) as T(i);
 cmsketch_rangecount 
---------------------
                1054
(1 row)

select cmsketch_rangecount(cmsketch(i),1,200) from generate_series(1,10000) as R(i);
 cmsketch_rangecount 
---------------------
                 225
(1 row)

select cmsketch_width_histogram(cmsketch(i), min(i), max(i), 4) from generate_series(1,10000) as R(i);
                                cmsketch_width_histogram                                
----------------------------------------------------------------------------------------
 [[1L, 2500L, 2526], [2501L, 5000L, 2515], [5001L, 7500L, 2517], [7501L, 10000L, 2518]]
(1 row)

select min(i),
       cmsketch_centile(cmsketch(i), 25, count(i)) AS quartile1, 
       cmsketch_centile(cmsketch(i), 50, count(i)) AS quartile2, 
       cmsketch_median(cmsketch(i), count(i)) AS median,
       cmsketch_centile(cmsketch(i), 75, count(i)) AS quartile3,
       max(i) 
  from generate_series(1,10000) as R(i);
 min | quartile1 | quartile2 | median | quartile3 |  max  
-----+-----------+-----------+--------+-----------+-------
   1 |      2487 |      4993 |   4993 |      7493 | 10000
(1 row)

select cmsketch_depth_histogram(cmsketch(i), 4) from generate_series(1,10000) as R(i);
                                             cmsketch_depth_histogram                                              
-------------------------------------------------------------------------------------------------------------------
 [[-9223372036854775807L, 2487, 2495], [2488, 4993, 2512], [4994, 7493, 2512], [7494, 9223372036854775807L, 2512]]
(1 row)

-- test for all-NULL column
select cmsketch_count(cmsketch(NULL), 5) from generate_series(1,10000) as R(i) where i < 0;
 cmsketch_count 
----------------
              0
(1 row)

